File: java/functions/ReturningValue.java
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

class ReturningValue {
  static boolean returningBoolean() {
    return true;
  }

  static int returningInteger() {
    return 7;
  }

  static String returningString() {
    return "hello";
  }

  public static void main(String[] args) {
    boolean b;
    int i;
    String s;

    b = returningBoolean();
    System.out.println(String.format("boolean:\nb = %s\n", b));

    i = returningInteger();
    System.out.println(String.format("integer:\ni = %d\n", i));

    s = returningString();
    System.out.println(String.format("string:\ns = %s\n", s));
  }
}

File: php/functions/returning_value.php
<?php
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

function returning_boolean(): bool
{
  return true;
}

function returning_integer(): int
{
  return 7;
}

function returning_string(): string
{
  return "hello";
}

$b = returning_boolean();
print("boolean:\n"
  . "b = {$b}\n\n");

$i = returning_integer();
print("integer:\n"
  . "i = {$i}\n\n");

$s = returning_string();
print("string:\n"
  . "s = {$s}\n\n");
